sklearn/decomposition/fastica_.py

```diff
@@ -563,8 +565,7 @@ def inverse_transform(self, X, copy=True):
         """
         check_is_fitted(self, 'mixing_')
 
-        if copy:
```

Comment:
Have replaced with `check_array`...! And yes I did this pre-emptively since there was a failure w.r.t to fast ica here -

```
======================================================================
ERROR: sklearn.tests.test_common.test_non_meta_estimators('FastICA', <class 'sklearn.decomposition.fastica_.FastICA'>)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/scikit-learn/sklearn/utils/testing.py", line 317, in wrapper
    return fn(*args, **kwargs)
  File "/scikit-learn/sklearn/utils/estimator_checks.py", line 669, in check_estimators_dtypes
    estimator.fit(X_train, y)
  File "/scikit-learn/sklearn/decomposition/fastica_.py", line 522, in fit
    self._fit(X, compute_sources=False)
  File "/scikit-learn/sklearn/decomposition/fastica_.py", line 478, in _fit
    compute_sources=compute_sources, return_n_iter=True)
  File "/scikit-learn/sklearn/decomposition/fastica_.py", line 300, in fastica
    X -= X_mean[:, np.newaxis]
TypeError: Cannot cast ufunc subtract output from dtype('float64') to dtype('int64') with casting rule 'same_kind'
```